[Android] MediaCodec: Add support for AV1 Dolby Vision codec selection

quietvoid opened the pull request:
## Description
For AV1 files with Dolby Vision stream side data, the profile must be specified as the mime type is not enough to query the correct MediaCodec codec.

Also, some devices support AV1 Dolby Vision on API version < 30, so we should attempt to use the profile constant regardless of the API version.

The changes bring support for AV1 Dolby Vision on devices that support it.

## Motivation and context
Being able to play Dolby Vision AV1 files.

## How has this been tested?
Runtime tested on Amazon FireTV Stick 4K Max.

Example of codec selection log: [kodi.txt](https://github.com/xbmc/xbmc/files/12284519/kodi.txt)

## What is the effect on users?
Should allow Dolby Vision playback in AV1 files.
No negative effect for existing behaviours.

Unfortunately `CAndroidUtils::SupportsMediaCodecMimeType("video/dolby-vision")` would likely always return support for HEVC codec, but I'm not sure if it's worth adding a function to also check if the profile is supported prior to setting the mime type.
As that's what the codec testing code already does, except it can't fallback to another mime type.
It's possible that playback of Dolby Vision in AV1 might not fallback to a HW decoder if MediaCodec only supports Dolby Vision in HEVC.

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

thexai commented on xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:
```diff
@@ -531,14 +531,9 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
-                  "CDVDVideoCodecAndroidMediaCodec::Open Dolby Vision playback support: "
-                  "Display: {}, MediaCodec: {}",
-                  displaySupportsDovi, mediaCodecSupportsDovi);
+        bool displaySupportsDovi, mediaCodecSupportsDovi;
```
Our code guidelines not allows declare multiple variables in same line.

```
bool displaySupportsDovi{false};
bool mediaCodecSupportsDovi{false};
```

thexai commented on xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:
```diff
@@ -634,6 +629,26 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
+        std::tie(displaySupportsDovi, mediaCodecSupportsDovi) =
+            CAndroidUtils::GetDolbyVisionCapabilities();
+
+        bool notHasHDR10fallback = (m_hints.dovi.dv_bl_signal_compatibility_id == 0 ||
```
```suggestion
        const bool notHasHDR10fallback = (m_hints.dovi.dv_bl_signal_compatibility_id == 0 ||
```

thexai commented on xbmc/windowing/android/AndroidUtils.cpp:
```diff
@@ -438,3 +438,14 @@ bool CAndroidUtils::SupportsMediaCodecMimeType(const std::string& mimeType)
+std::pair<bool, bool> CAndroidUtils::GetDolbyVisionCapabilities()
+{
+  bool displaySupportsDovi = CAndroidUtils::GetDisplayHDRCapabilities().SupportsDolbyVision();
+  bool mediaCodecSupportsDovi = CAndroidUtils::SupportsMediaCodecMimeType("video/dolby-vision");
```
```suggestion
  const bool mediaCodecSupportsDovi = SupportsMediaCodecMimeType("video/dolby-vision");
```
Class name prefix is not needed since we are in the same class
Same in `displaySupportsDovi`

thexai reviewed (commented):
Looks good in general, only some minor formatting things:

quietvoid committed:
[Android] MediaCodec: Add support for AV1 Dolby Vision codec selection

For AV1 files with Dolby Vision stream side data, the profile must be specified
as the mime type is not enough to query the correct MediaCodec codec.

Also, some devices support AV1 Dolby Vision on API version < 30, so we should
attempt to use the profile constant regardless of the API version.

garbear commented on xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:
```diff
@@ -634,6 +631,28 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
+      if (m_hints.hdrType == StreamHdrType::HDR_TYPE_DOLBYVISION && m_hints.dovi.dv_profile == 10)
+      {
+        bool displaySupportsDovi{false};
+        bool mediaCodecSupportsDovi{false};
```
You don't need to define here, declaring is enough because this can't be undefined after the next line, and the choice of initializer value here (true/false) has no effect.
